client/client.go

```diff
@@ -293,14 +294,22 @@ func (m *K8sClient) WaitPodsReady(ns string, rcd *ReadyCheckData, expectedPodCou
 					break
 				}
 				for _, c := range pod.Status.Conditions {
-					if c.Type == v1.ContainersReady && c.Status == "False" {
+					if c.Type == v1.ContainersReady && c.Status != "True" {
 						log.Debug().Str("Text", c.Message).Msg("Pod condition message")
 						allReady = false
 					}
 				}
 			}
+
 			if allReady {
-				return nil
+				readyCount++
+				// wait for it to be ready 3 times since there is no good way to know if an old pod
+				// was present but not yet decommisiond during a rollout
+				// usually there is just a very small blip that we can run into this and this will
+				// prevent that from happening
```

Comment:
It might be possible. We would have to dig through the old chart to see what pods are changing and then do a comparison to the new chart by digging through it, use that info to build a map of what pods we think are changing and unset the app label on those. Wouldn't affect this particular loop in the code since this catches more than just issues with the replaceHelm call but definitely a good thing to look into in the future.